### PR TITLE
[protocol handler]: add ability to open gists

### PR DESCRIPTION
**Pull request by legomushroom**
The PR adds protocol handler and to routes to open gists:

  - `Using gist Id`: vscode://vsls-contrib.gistfs/open-gist?id=b01737ed99192dab436adea1d6d92975
  - `Using gist URL`: vscode://vsls-contrib.gistfs/open-gist-url?url=https://gist.github.com/legomushroom/b01737ed99192dab436adea1d6d92975

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## v0.0.16 (12/08/2019)
```
Comment (lostintangent) [inline CHANGELOG.md]: I haven’t shipped 0.0.15 yet, so can you move this to the 0.0.15 section?

File: src/protocolHandler.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export class GistPadProtocolHandler implements vscode.UriHandler {
+  public async handleUri(uri: vscode.Uri): Promise<void> {
+    switch (uri.path) {
+      case '/open-gist': {
+        const openGist = openGistFactory('id');
+
+        return await openGist(uri);
+      }
+
+      case '/open-gist-url': {
```
Comment (lostintangent) [inline src/protocolHandler.ts]: Should we just have “open-gist” and allow it to specify an ID or URL? Then we could allow URL to take precedence if both are specified.

File: src/protocolHandler.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    const propName = (paramName === 'id')
+      ? 'gistId'
+      : 'gistUrl'
+
+    await vscode.commands.executeCommand(CommandId.openGist, { [propName]: gistParam });
+  }
+}
+
+export class GistPadProtocolHandler implements vscode.UriHandler {
```
Comment (lostintangent) [inline src/protocolHandler.ts]: Should we add another query option to specify opening it as a workspace?

File: CHANGELOG.md (hunk 1 above)
Comment (legomushroom) [reply CHANGELOG.md]: Done.

File: src/protocolHandler.ts (hunk 2 above)
Comment (legomushroom) [reply src/protocolHandler.ts]: Done.

File: src/protocolHandler.ts (hunk 3 above)
Comment (legomushroom) [reply src/protocolHandler.ts]: Done.

**Pull request merged**